rustls/src/msgs/deframer.rs

```diff
@@ -524,6 +528,41 @@ mod tests {
         include_bytes!("../testdata/deframer-invalid-version.bin");
     const INVALID_LENGTH_MESSAGE: &[u8] = include_bytes!("../testdata/deframer-invalid-length.bin");
 
+    // buffered version to ease testing
+    #[derive(Default)]
+    struct MessageDeframer {
```

Comment:
Maybe give this a different name, like `BufferedDeframer`? Also, probably makes sense to move `input_bytes()` and `input_bytes_concat()` into to this type's `impl`.